### PR TITLE
[CORE-1371] Disallow changing project or pipeline name in pachctl edit pipeline

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

File: src/server/pps/cmds/cmds.go [hunk 1]
```diff
@@ -881,6 +881,27 @@ All jobs created by a pipeline will create commits in the pipeline's output repo
...
+			// then it is considered unchanged.
+			project := request.Pipeline.GetProject()
+			projectName := project.GetName()
+			if projectName == "" {
+				if project == nil {
+					project = new(pfs.Project)
+				}
+				project.Name = pipelineInfo.Pipeline.GetProject().GetName()
+				request.Pipeline.Project = project
+			} else if projectName != pipelineInfo.Pipeline.GetProject().GetName() {
```
Comment (albscui) [inline src/server/pps/cmds/cmds.go:894]: Can we move the exceptional cases up?
```Go
if projectName != pipelineInfo.Pipeline.GetProject().GetName() {
    return errors.New("may not change project name")
}
if pipelineName != pipelineInfo.Pipeline.GetName() {
     return errors.New("may not change pipeline name")
}

// business logic ...
```

Comment (albscui) [reply src/server/pps/cmds/cmds.go:894]: Also given that we don't allow the pipeline/project name to change, I think we can safely assign it to the request without conditionals.

**Review by acohen4 — APPROVED** [review]
LGTM

File: src/server/pps/cmds/cmds.go (hunk 1 above)
Comment (acohen4) [reply src/server/pps/cmds/cmds.go:894]: +1

**Pull request merged**